spectral-extraction: API (#1554)
spacetelescope/jdaviz: docs/specviz2d/plugins.rst

The Spectral Extraction plugin exposes `specreduce <https://specreduce.readthedocs.io>`_
methods for tracing, background subtraction, and spectral extraction from 2D spectra.

To interact with the plugin via the API in a notebook, access the plugin object via::

>   sp_ext = viz.app.get_tray_item_from_name('spectral-extraction')

Reviewer: What do you think of adding a `viz.spectral_extraction` or `viz.spectral_extraction_plugin` attribute to the Specviz2D helper that is set to this, so that the user doesn't have to go through the app-level incantation?

Author: How would you feel about re-considering #1401?  It's _slightly_ more verbose than what you suggest, but I think more friendly than the current app-level method.

Reviewer: Hmm....I'm fine with thinking about this harder in the context of 1401 and not holding up this PR.

Author: I agree that documenting any methods on `viz.app` is a bit of a slippery slope, but it is one we've already started sliding down...

A preview of the trace will update in real time in the 2D spectrum viewer.

To export and access the specreduce Trace object defined in the plugin, call :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_trace`::

  trace = sp_ext.export_trace()

> To import the parameters from a Trace object (either a new Trace object, or an exported one modified in the notebook) into the plugin, call :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.import_trace`::

Reviewer 2: ```suggestion
To import the parameters from a specreduce Trace object, whether it's new or was exported and modified in the notebook, call:meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.import_trace`::
```
Consistency with the analogous line in the other sections.

To refine the trace based on the background-subtracted image, return
to the Trace step and select the exported background-subtracted image as input. 

To export and access the specreduce Background object defined in the plugin, call :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_bg`::

  bg = sp_ext.export_bg()

> To access the background image or background-subtracted image as a specutils Spectrum1D object, call :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_bg_img` or :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_bg_img`, respectively.

Reviewer 2: I think this one should still work since it's `specutils` and not `specreduce`. It's used on other pages, too.

> To access the background image or background-subtracted image as a specutils Spectrum1D object, call :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_bg_img` or :meth:`~jdaviz.configs.specviz2d.plugins.spectral_extraction.spectral_extraction.SpectralExtraction.export_bg_img`, respectively.

Author: woops, good catch.  Fixed!
